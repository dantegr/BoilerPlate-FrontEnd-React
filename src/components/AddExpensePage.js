import React from 'react';

const AddExpensePage = () => (
  <div>this is add expense component</div>
);

export default AddExpensePage;