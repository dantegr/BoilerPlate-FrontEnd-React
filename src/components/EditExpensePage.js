import React from 'react';

const EditExpensePage = () => (
  <div>this is edit expense component</div>
);

export default EditExpensePage;