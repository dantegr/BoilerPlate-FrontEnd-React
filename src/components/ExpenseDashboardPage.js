import React from 'react';

const ExpenseDashboardPage = () => (
  <div>this is dahborad component</div>
);

export default ExpenseDashboardPage;