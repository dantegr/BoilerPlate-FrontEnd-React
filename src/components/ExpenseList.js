import React from 'react';
import { connect } from 'react-redux';


const ExpenseList = (props) => (
  <div>
    <h1>Expense List</h1>
    {props.filter.text}
    {props.expenses.length}
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: state.expenses,
    filter: state.filter
  };
};

export default connect(mapStateToProps)(ExpenseList);
