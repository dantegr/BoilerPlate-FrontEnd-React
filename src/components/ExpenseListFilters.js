import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate} from '../actions/filters';

class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  };

  onDatesChange = ({ startDate, endDate}) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  };

  onFocusChange = (calendarFocused) => {
    this.setState(() => ({calendarFocused}));
  };

  render() {
    return (
      <div>
      <input type="text" value={this.props.filter.text} onChange={(e) => {
        this.props.dispatch(setTextFilter(e.target.value));
      }}/>
      <select
      value= {this.props.filter.sortBy} 
      onChange={(e) => {
        if (e.target.value === "date"){
          this.props.dispatch(sortByDate());
        } else if (e.target.value === "amount"){
          this.props.dispatch(sortByAmount());
        }
      }}>
        <option value="date">Date</option>
        <option value="amount">Amount</option>
      </select>
      <DateRangePicker
        startDate={this.props.filter.startDate} // momentPropTypes.momentObj or null,
        //startDateId="your_unique_start_date_id" // PropTypes.string.isRequired,
        endDate={this.props.filter.endDate} // momentPropTypes.momentObj or null,
        //endDateId="your_unique_end_date_id" // PropTypes.string.isRequired,
        onDatesChange={this.onDatesChange} // PropTypes.func.isRequired,
        focusedInput={this.state.calendarFocused} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
        onFocusChange={this.onFocusChange} // PropTypes.func.isRequired,
        numberOfMonths={1}
        isOutsideRange={() => false}
        showClearDates={true}
      />
    </div>

    )


  }


}



const mapStateToProps = (state) => {
  return {
    filter: state.filter
  };
};

export default connect(mapStateToProps)(ExpenseListFilters);