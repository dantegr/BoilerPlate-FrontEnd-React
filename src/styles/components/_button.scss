.button {
  background: $dark-green;
  color: white;
  border: none;
  padding: $s-size;
  font-weight: 300;
  font-size: $font-size-large;
}