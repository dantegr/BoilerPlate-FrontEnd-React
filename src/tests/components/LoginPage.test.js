import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should correctly render LoginPage component', () => {
  const wrapper = shallow(<LoginPage/>);
  expect(wrapper).toMatchSnapshot();
});