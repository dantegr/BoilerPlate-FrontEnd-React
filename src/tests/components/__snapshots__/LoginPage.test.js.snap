// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render LoginPage component 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expenser
    </h1>
    <p>
      Manage your expenses the right way!
    </p>
    <button
      className="button"
    >
      Login with Google!
    </button>
  </div>
</div>
`;
